tests/v2/utils/library/project_test_utils.go

```diff
@@ -0,0 +1,192 @@
+		errorString = append(errorString, commonUtils.LogErrorMessage(fmt.Sprintf("Project array compare failed.")))
+		for _, project := range parserProjects {
+			if testProject, found := getSchemaProject(devfile.SchemaDevFile.Projects, project.Name); found {
+				if !cmp.Equal(project, *testProject) {
+					// Write out the failing project to a file, once as expected by the test, and a second as returned by the parser
+					parserFilename := commonUtils.AddSuffixToFileName(devfile.FileName, "_"+project.Name+"_Parser")
+					testFilename := commonUtils.AddSuffixToFileName(devfile.FileName, "_"+project.Name+"_Test")
+					commonUtils.LogInfoMessage(fmt.Sprintf(".......marshall and write devfile %s", parserFilename))
+					c, err := yaml.Marshal(project)
+					if err != nil {
+						errorString = append(errorString, commonUtils.LogErrorMessage(fmt.Sprintf(".......marshall devfile %s", parserFilename)))
+					} else {
+						err = ioutil.WriteFile(parserFilename, c, 0644)
+						if err != nil {
+							errorString = append(errorString, commonUtils.LogErrorMessage(fmt.Sprintf(".......write devfile %s", parserFilename)))
+						}
+					}
+					commonUtils.LogInfoMessage(fmt.Sprintf(".......marshall and write devfile %s", testFilename))
+					c, err = yaml.Marshal(testProject)
+					if err != nil {
+						errorString = append(errorString, commonUtils.LogErrorMessage(fmt.Sprintf(".......marshall devfile %s", testFilename)))
+					} else {
+						err = ioutil.WriteFile(testFilename, c, 0644)
+						if err != nil {
+							errorString = append(errorString, commonUtils.LogErrorMessage(fmt.Sprintf(".......write devfile %s", testFilename)))
+						}
+					}
+					errorString = append(errorString, commonUtils.LogErrorMessage(fmt.Sprintf("Project %s did not match, see files : %s and %s", project.Name, parserFilename, testFilename)))
+				} else {
+					commonUtils.LogInfoMessage(fmt.Sprintf(" --> Project matched : %s", project.Name))
+				}
+			} else {
+				errorString = append(errorString, commonUtils.LogErrorMessage(fmt.Sprintf("Project from parser not known to test - id : %s ", project.Name)))
+			}
+		}
+		// Check test does not include projects which the parser did not return
+		for _, project := range devfile.SchemaDevFile.Projects {
+			if _, found := getSchemaProject(parserProjects, project.Name); !found {
+				errorString = append(errorString, commonUtils.LogErrorMessage(fmt.Sprintf("Project from test not returned by parser : %s ", project.Name)))
+			}
+		}
+	} else {
+		commonUtils.LogInfoMessage(fmt.Sprintf("Project structures matched"))
+	}
+
+	var err error
+	if len(errorString) > 0 {
+		err = errors.New(fmt.Sprint(errorString))
+	}
+	return err
+}
+
+// VerifyStarterProjects verifies starter projects returned by the parser are the same as those saved in the devfile schema
+func VerifyStarterProjects(devfile *commonUtils.TestDevfile, parserStarterProjects []schema.StarterProject) error {
```

Comment:
shouldnt this be called from `verify`

Reply:
Yes, lol, not sure how I lost that code. Too many branches!